### CI note: E2E `skill-eval` failure is an Anthropic API outage, not a regression

The red **E2E Tests** job is the two `test/e2e/skill-eval.test.ts` LLM-eval cases (`claude-sonnet-4-6`/`claude-opus-4-6 meets threshold`) failing with:

```
[planner] API error: Invalid response body while trying to fetch https://api.anthropic.com/v1/messages: Premature close
AssertionError: expected 0 to be greater than or equal to 0.75
```

These call `api.anthropic.com` directly; the `Premature close` → score `0` is an upstream API outage. All **126** non-LLM E2E tests pass, and this PR only touches the skill-file *write* mechanism (`agent-skills.ts`), which cannot affect eval scores. Re-running once the API recovers should clear it.

### Pushed: test hardening (commit `3393368`)
An adversarial self-review found the original atomic-write test was tautological — it passed even when `atomicWriteFile` was reverted to a plain `writeFile` (verified by mutation). Replaced it with real guards:
- asserts every `writeFile` targets a `.tmp` staging path and the dest is published via `rename` (fails under a non-atomic mutation);
- a rollback test covering the previously-untested catch/cleanup branch (rename fails → returns `null`, temp cleaned up, no partial `SKILL.md`).

--- a/test/lib/agent-skills.test.ts
+++ b/test/lib/agent-skills.test.ts
@@ -6,7 +6,7 @@
  */
 
 import { chmodSync, existsSync, mkdirSync, readdirSync, rmSync } from "node:fs";
-import { readFile, rename, writeFile } from "node:fs/promises";
+import { readFile, rename, rm, writeFile } from "node:fs/promises";
 import { join } from "node:path";
 import { afterEach, beforeEach, describe, expect, test, vi } from "vitest";
 import {
@@ -199,6 +199,22 @@ describe("agent-skills", () => {
       );
       expect(leftovers).toEqual([]);
       expect(existsSync(join(skillDir, "SKILL.md"))).toBe(false);
+    });
+
+    test("still fails to null when temp cleanup also fails after a rename error", async () => {
+      mkdirSync(join(testDir, ".agents"), { recursive: true });
+
+      // Both the rename and the subsequent best-effort `rm` of the temp file
+      // fail. The cleanup error must be swallowed (captured, not thrown) so the
+      // original rename error is what propagates — exercising the inner
+      // catch of atomicWriteFile.
+      vi.mocked(rename).mockRejectedValueOnce(new Error("rename failed"));
+      vi.mocked(rm).mockRejectedValueOnce(new Error("cleanup failed"));
+
+      const result = await installAgentSkills(testDir);
+
+      expect(result).toBeNull();
+      expect(vi.mocked(rm)).toHaveBeenCalled();
     });
 
     test("installs to both ~/.agents/ and ~/.claude/ when both roots exist", async () => {

